### PR TITLE
libc:machine:xtensa:add xtensa libc implement

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
## Summary

Add xtensa arch libc implement

## Impact
No.

## Testing

Testing pass on memcpy_test, memset_test.

### CODE REVIEW

File: libs/libc/machine/xtensa/arch_memcpy.S [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+ * libs/libc/machine/xtensa/arch_memcpy.S
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http:#www.apache.org/licenses/LICENSE-2.0
```
Comment (Ouss4) [inline libs/libc/machine/xtensa/arch_memcpy.S:11]:
```suggestion
 *   http://www.apache.org/licenses/LICENSE-2.0
```

**Comment by xiaoxiang781216** [comment]
@Ouss4 could you review the update again?

File: libs/libc/machine/xtensa/arch_setjmp.S_bk [hunk 2]
```diff
@@ -0,0 +1,374 @@
+/****************************************************************************
+ * libs/libc/machine/xtensa/arch_setjmp.S
```
Comment (Ouss4) [inline libs/libc/machine/xtensa/arch_setjmp.S_bk:2]: Looks like this files sneaked in with a `git add .`
(This looks like what was supposed to be a backup file)

**Comment by zhuyanlinzyl** [comment]
@Ouss4  Thansk. Please review again

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**